### PR TITLE
pre_release.timesensitive: Adding status_server uri and using custom TestSuite

### LINKED ISSUE #4153: [Bug] Avocado crash with TypeError
With the following change on the time-sensitive job Avocado crashes:

```python
diff --git a/selftests/pre_release/jobs/timesensitive.py b/selftests/pre_release/jobs/timesensitive.py
index a9fbebcd..456719aa 100755
--- a/selftests/pre_release/jobs/timesensitive.py
+++ b/selftests/pre_release/jobs/timesensitive.py
@@ -4,6 +4,7 @@ import os
 import sys
 
 from avocado.core.job import Job
+from avocado.core.suite import TestSuite
 
 THIS_DIR = os.path.dirname(os.path.abspath(__file__))
 ROOT_DIR = os.path.dirname(os.path.dirname(os.path.dirname(THIS_DIR)))
@@ -19,6 +20,7 @@ CONFIG = {
 
 
 if __name__ == '__main__':
-    with Job(CONFIG) as j:
+    suite = TestSuite.from_config(CONFIG)
+    with Job(CONFIG, [suite]) as j:
         os.environ['AVOCADO_CHECK_LEVEL'] = '3'
         sys.exit(j.run())
```

Crash:

```
[wrampazz@wrampazz avocado.dev]$ selftests/pre_release/jobs/timesensitive.py
JOB ID     : 5c1cf735be942802efc655a82ec84e46c1301080
JOB LOG    : /home/wrampazz/avocado/job-results/job-2020-08-27T16.12-5c1cf73/job.log

Avocado crashed: TypeError: expected str, bytes or os.PathLike object, not NoneType
Traceback (most recent call last):

  File "/home/wrampazz/src/avocado/avocado.dev/avocado/core/job.py", line 605, in run_tests
    summary |= suite.run(self)

  File "/home/wrampazz/src/avocado/avocado.dev/avocado/core/suite.py", line 266, in run
    return self.runner.run_suite(job, self)

  File "/home/wrampazz/src/avocado/avocado.dev/avocado/plugins/runner_nrunner.py", line 237, in run_suite
    loop.run_until_complete(asyncio.wait_for(asyncio.gather(*workers),

  File "/usr/lib64/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()

  File "/usr/lib64/python3.8/asyncio/tasks.py", line 455, in wait_for
    return await fut

  File "/home/wrampazz/src/avocado/avocado.dev/avocado/core/task/statemachine.py", line 155, in run
    await self.start()

  File "/home/wrampazz/src/avocado/avocado.dev/avocado/core/task/statemachine.py", line 113, in start
    start_ok = await self._spawner.spawn_task(runtime_task)

  File "/home/wrampazz/src/avocado/avocado.dev/avocado/plugins/spawners/process.py", line 29, in spawn_task
    runtime_task.spawner_handle = await asyncio.create_subprocess_exec(

  File "/usr/lib64/python3.8/asyncio/subprocess.py", line 236, in create_subprocess_exec
    transport, protocol = await loop.subprocess_exec(

  File "/usr/lib64/python3.8/asyncio/base_events.py", line 1630, in subprocess_exec
    transport = await self._make_subprocess_transport(

  File "/usr/lib64/python3.8/asyncio/unix_events.py", line 197, in _make_subprocess_transport
    transp = _UnixSubprocessTransport(self, protocol, args, shell,

  File "/usr/lib64/python3.8/asyncio/base_subprocess.py", line 36, in __init__
    self._start(args=args, shell=shell, stdin=stdin, stdout=stdout,

  File "/usr/lib64/python3.8/asyncio/unix_events.py", line 789, in _start
    self._proc = subprocess.Popen(

  File "/usr/lib64/python3.8/subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,

  File "/usr/lib64/python3.8/subprocess.py", line 1637, in _execute_child
    self.pid = _posixsubprocess.fork_exec(

TypeError: expected str, bytes or os.PathLike object, not NoneType

Please include the traceback info and command line used on your bug report
Report bugs visiting https://github.com/avocado-framework/avocado/issues/new
```

### PR DESCRIPTION
**Pull request by beraldoleal**
This fixes #4153

Signed-off-by: Beraldo Leal <bleal@redhat.com>

### CODE REVIEW

File: selftests/pre_release/jobs/timesensitive.py [hunk 1]
```diff
@@ -14,11 +15,13 @@
     'run.references': [os.path.join(ROOT_DIR, 'selftests', 'unit'),
                        os.path.join(ROOT_DIR, 'selftests', 'functional')],
     'filter.by_tags.tags': ['parallel:1'],
+    'nrunner.status_server_uri': '127.0.0.1:9999',
```
Comment (pevogam) [inline selftests/pre_release/jobs/timesensitive.py:18]: How about using the default 8888 used elsewhere? Just a bit less entropy in the code this way as all included values are the same.

File: selftests/pre_release/jobs/timesensitive.py [hunk 2]
```diff
@@ -14,11 +15,13 @@
...
     'nrunner.max_parallel_tasks': 1,
     }
 
 
 if __name__ == '__main__':
-    with Job(CONFIG) as j:
+    suite = TestSuite.from_config(CONFIG)
+    with Job(CONFIG, [suite]) as j:
```
Comment (pevogam) [inline selftests/pre_release/jobs/timesensitive.py:25]: I guess `Job.from_config(CONFIG)` is not possible here? It seems the test suite and job use the same config.

**Review by pevogam — APPROVED** [review]
Good fix!

File: selftests/pre_release/jobs/timesensitive.py (hunk 2 above)
Comment (beraldoleal) [reply selftests/pre_release/jobs/timesensitive.py:25]: Yes, in this case they are using, we could have something like:

```python
with Job.from_config(CONFIG) as j:
```

File: selftests/pre_release/jobs/timesensitive.py (hunk 1 above)
Comment (beraldoleal) [reply selftests/pre_release/jobs/timesensitive.py:18]: Good catch, yes, the examples are using 9999 but the default is 8888. will change.

**Comment by beraldoleal** [comment]
@pevogam force pushed here with the changes.

File: selftests/pre_release/jobs/timesensitive.py (hunk 2 above)
Comment (pevogam) [reply selftests/pre_release/jobs/timesensitive.py]: In this case it could be simpler to parse, as long as we get the same end behavior.

Comment (beraldoleal) [reply selftests/pre_release/jobs/timesensitive.py]: Yes, when creating a job using from_config() and if there is no suite's config, this method then will create one test suite with the same config. `len(job.test_suites) == 1` in this case.

File: selftests/pre_release/jobs/timesensitive.py (hunk 1 above)
Comment (clebergnu) [reply selftests/pre_release/jobs/timesensitive.py]: > How about using the default 8888 used elsewhere? Just a bit less entropy in the code this way as all included values are the same.

+1.  But the fact is, there's some smelly here.  It should not be necessary to set a status server given that it's registered.  I've created an issue to track it here: https://github.com/avocado-framework/avocado/issues/4157

**Comment by clebergnu** [comment]
Actually, it seems like there's *another* problem that get's triggered here.  @beraldoleal @willianrampazzo @pevogam are you able to actually run the job? Mine its stuck and the reason is:

```
Process Process-1:
Traceback (most recent call last):
  File "/usr/lib64/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/usr/lib64/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "/home/cleber/src/avocado/avocado/avocado/core/nrunner.py", line 476, in _run_unittest
    suite = unittest.TestLoader().loadTestsFromName(unittest_name)
  File "/usr/lib64/python3.7/unittest/loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
ValueError: Empty module name
```

We need to pass a relative location to the tests.  With the absolute path, no tests will by found by Python's `unittest.loader` module, because it looks for module names.

**Comment by clebergnu** [comment]
> Holding it for a moment while we discuss [#4154 (comment)](https://github.com/avocado-framework/avocado/pull/4154#issuecomment-682208873)

Actually, it's a different issue, so I'm merging this so we can focus on the other problem.

**Comment by pevogam** [comment]
> Actually, it seems like there's _another_ problem that get's triggered here. @beraldoleal @willianrampazzo @pevogam are you able to actually run the job? Mine its stuck and the reason is:
> 
> ```
> Process Process-1:
> Traceback (most recent call last):
>   File "/usr/lib64/python3.7/multiprocessing/process.py", line 297, in _bootstrap
>     self.run()
>   File "/usr/lib64/python3.7/multiprocessing/process.py", line 99, in run
>     self._target(*self._args, **self._kwargs)
>   File "/home/cleber/src/avocado/avocado/avocado/core/nrunner.py", line 476, in _run_unittest
>     suite = unittest.TestLoader().loadTestsFromName(unittest_name)
>   File "/usr/lib64/python3.7/unittest/loader.py", line 154, in loadTestsFromName
>     module = __import__(module_name)
> ValueError: Empty module name
> ```
> 
> We need to pass a relative location to the tests. With the absolute path, no tests will by found by Python's `unittest.loader` module, because it looks for module names.

I am lagging a bit with the versions and my branches are some hours behind but maybe the rest could confirm.

**Pull request merged**